Automattic/wp-calypso
# Stats: Update code structure for the insights page

## Proposed Changes

* Organizes code structure for section components present on the insights page.

## Why are these changes being made?
* This is part of an ongoing effort to pay off tech debt in Stats code organization: pejTkB-1tb-p2.

## Testing Instructions



* Open this PR's live branch and navigate to the insights dashboard (`/stats/insights/example.com`).
* Ensure the page works identically as before. No business logic was changed. 

## Pre-merge Checklist
